ni/abc.py

```diff
@@ -63,6 +63,11 @@ def contrib_auth_token(self):
         raise NotImplementedError
 
     @abc.abstractmethod
+    def secret_token(self):
+        """Return the authorization token for the contribution host."""
+        raise NotImplementedError
```

Comment:
Let's have this return `None` for cases where this isn't provided as this is a nicety and if for some reason it isn't available it isn't the end of the world.